orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -414,14 +414,15 @@ def forward(ctx, *inputs, **kwargs):
                 return user_outputs[0] if len(user_outputs) == 1 else user_outputs
 
             @staticmethod
-            def backward(ctx, *grad_output):
+            def backward(ctx, *grad_outputs):
                 '''Performs backward pass based on grad wrt module output
                 '''
 
                 # Use IO binding
                 # Push user output grads to ONNX backend.
                 backward_grad_output_ortvalue = []
-                for grad_output in grad_output[:len(self._onnx_graphs_info.backward_output_grad_names)]:
+                for _, i in self._onnx_graphs_info.backward_output_grad_names_map.items():
```

Comment:
Let's add a comment here 
backward_output_grad_names_map only contains the subset of module outputs that need a gradient, we filter out the invalid entries in 'grad_outputs'